### ISSUE TITLE
Cooperative play between Sorcerer and fighter

### ISSUE THREAD

**Issue by marouen-lamiri** [marouen-lamiri/Second-Soul#73]
As a user, I want to be able to participate in another player’s game

[SP: 20]
[Priority: 1]

**Comment by smileyWolf** [comment]
#### Black-Box Testing

| Tests # | Inputs | Expected Output | Status |
| --- | --- | --- | --- |
| 1. Connect players | Open StartScreen scene in two game instances. Create a Fighter in one, and a Sorcerer in the other | <b>Test that every previous feature works when playing over the network</b> and that <b>every action done in one instance is also visible in the other</b>. | Pass |

**Comment by Kishor0122** [comment]
Nicely done! Connecting to another person's game works; the server user plays as the fighter and the client user plays as the sorcerer. Both are then connected to the same game and are playing together. Minor things to fix for later iterations: other user's movement animation is missing, both characters should spawn (and re-spawn) at the same location, sorcerer attacking enemies should aggro them to the fighter. Looks good for now though, signing off.

**Comment by Kishor0122** [comment]
Correction, movement animation works

**Comment by nico3865** [comment]
<b>Task Breakdown (actual):</b>

--dontdestroyOnLoad both characters
	we Network.Instantiate them both on the server side, 
	.find them on both sides
	DontDestroyOnLoad them on both sides too
	... this solved the issue of:
		not being able to run the game scene without a player
		not being able to save a character over during the transfer from menu scene to game scene
--making the server wait longer (700 frames)
	this solved the issue of:
		the map being Network.Instantiated and received on the client side WHILE IT WAS STILL IN THE MENU SCENE --> thus the whole map was not kept during the scene transfer
--passing the position of the sorcerer after the map was created, because ... because the map updates its position
--focus correct player 
--enemy network instantiation
	--network instantiate positions?
	--watch attack anim
	--watch die / health.
	--mouse over to see their health (fixed)
	--enemy to tell the sorcerer in the client game instance to acquire it as a target. I'm putting that on my list too, tell me if you see any relevant (files:lines)
// NORMAL -- FIX: why is the sorcerer dying on the client when the fighter dies on the server?
--sorcerer --> implement rpcs for watching vars
	--basic attack like fighter.
	--spell cast (lose energy as it's done?) -- that's the fireball.
	// gain energy? already done
	--energy: You are right about the RPC call for decreasing the energy of the fighter, --> the sorcerer would have to make an RPC call to remove energy from the fighter, and then the energy RPC call currently in place will broadcast that back to the fighter in the client game instance.
	--1- make Sorcerer be able to make enemy a target. (you will see this if there's a health bar) 
	--2- make Sorcerer basic attack work
	3- fireball
		collision sphere?
		flammes?
	network instantiate the fireball -- in fireballSkill.cs line 48
		basically like an attack animation for the fighter.
		ie redo the fireball anim but don't redo the damage on the other instance.
fix miniMap for client.
--destroy player.
--clean commented ou code
--additional animations
	--walk
	--idle
--refactor enemyNetworkScript / player / sorcerer, to have a parent network script class.
--allow to toggle the chat window on / off.
--place the chat window so it looks good, like on the server instance (there it looks fine)
// Capitalize all method names. NOPE it's FINE.
